src/services/BlockchainScanner.ts

```diff
@@ -44,25 +59,53 @@ export abstract class BlockchainScanner {
     return {fromBlock, toBlock, success, synchronized: toBlock < fromBlock + this.step};
   }
 
-  private async getToBlock(fromBlock: number): Promise<number> {
+  setChainId(chainId: ChainsIds) {
+    this.chainId = chainId;
+  }
+
+  setProvider() {
+    if (!this.chainId) {
+      this.logger.error('[UniswapPoolScanner] chainId not set');
+      return;
+    }
+
+    if (!this.settings.blockchains[this.chainId].providerUrl.join('')) {
+      this.logger.error(`[UniswapPoolScanner] ${this.chainId}: provider URL not set`);
+      return;
+    }
+
+    this.provider = <StaticJsonRpcProvider>this.blockchainProviderRepository.get(this.chainId);
+    this.startBlock = this.settings.dexes.uniswap[this.chainId]?.startBlock || 0;
+    this.step = this.settings.dexes.uniswap[this.chainId]?.agentStep || 0;
+  }
+
+  private async getToBlock(fromBlock: number): Promise<number | undefined> {
     if (!this.provider) throw new Error('[BlockchainScanner] getToBlock: no provider');
 
     const latestBlock = await this.provider.getBlockNumber();
     const stepBlock = fromBlock + this.step;
+    if (stepBlock > latestBlock) {
```

Comment:
```
const blocksToScan = latestBlock - fromBlock;

if (blocksToScan < this.step) -> we do not scan
else return Math.trim(blocksToScan / this.step) * this.step;
```

this should be the way of calculating the range of `from - to`

Reply:
You mean Math.trunc, right?
But in that case, let say that blocksToScan = 100, step = 20, it will return 100, instead of 20